src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+            }
+
+            protected override int GetNearmostParentJumpStatementRawKind(IfStatementSyntax ifStatement)
+            {
+                foreach (var node in ifStatement.Ancestors())
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.SwitchSection:
+                            return (int)SyntaxKind.BreakStatement;
+
+                        case SyntaxKind.LocalFunctionStatement:
+                        case SyntaxKind.SetAccessorDeclaration:
+                        case SyntaxKind.GetAccessorDeclaration:
+                        case SyntaxKind.AddAccessorDeclaration:
+                        case SyntaxKind.RemoveAccessorDeclaration:
+                        case SyntaxKind.MethodDeclaration:
+                        case SyntaxKind.ConstructorDeclaration:
+                        case SyntaxKind.DestructorDeclaration:
+                        case SyntaxKind.OperatorDeclaration:
+                        case SyntaxKind.ConversionOperatorDeclaration:
+                        case SyntaxKind.AnonymousMethodExpression:
+                        case SyntaxKind.SimpleLambdaExpression:
+                        case SyntaxKind.ParenthesizedLambdaExpression:
+                            return (int)SyntaxKind.ReturnStatement;
+
+                        case SyntaxKind.DoStatement:
+                        case SyntaxKind.WhileStatement:
+                        case SyntaxKind.ForStatement:
+                        case SyntaxKind.ForEachStatement:
+                        case SyntaxKind.ForEachVariableStatement:
+                            return (int)SyntaxKind.ContinueStatement;
+                    }
+                }
+
+                return -1;
+            }
+
+            protected override (SyntaxNode first, SyntaxNode last) GetIfBodyStatementRange(IfStatementSyntax ifStatement)
+            {
+                var statement = ifStatement.Statement;
+                return (statement, statement);
+            }
+
+            protected override IEnumerable<(SyntaxNode first, SyntaxNode last)> GetSubsequentStatementRange(IfStatementSyntax ifStatement)
+            {
+                StatementSyntax innerStatement = ifStatement;
+                foreach (var node in ifStatement.Ancestors())
+                {
+                    var nextStatement = innerStatement.GetNextStatement();
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.Block:
+                            if (nextStatement != null) yield return (nextStatement, ((BlockSyntax)node).Statements.Last());
+                            break;
+                        case SyntaxKind.SwitchSection:
+                            if (nextStatement != null) yield return (nextStatement, ((SwitchSectionSyntax)node).Statements.Last());
+                            yield break;
```

Comment:
needs def explanations of why this is yield break (whereas it's just 'break' above).

Reply:
(resolved)